phpunit/fonts-library/class-wp-font-family-test.php

```diff
@@ -0,0 +1,182 @@
+<?php
+/**
+ * Tests for Font Family Utils class
+ *
+ * @package Gutenberg
+ */
+
+class WP_Font_Family_Test extends WP_UnitTestCase {
+
+	const FONT_DATA_1 = array(
+		'name'       => 'Piazzolla',
+		'slug'       => 'piazzolla',
+		'fontFamily' => 'Piazzolla',
+		'fontFace'   => array(
+			array(
+				'fontFamily' => 'Piazzolla',
+				'fontStyle'  => 'italic',
+				'fontWeight' => '400',
+				'src'        => 'http://fonts.gstatic.com/s/piazzolla/v33/N0b72SlTPu5rIkWIZjVgI-TckS03oGpPETyEJ88Rbvi0_TzOzKcQhZqx3gX9BRy5m5M.ttf',
+			),
+		),
+	);
+
+	const FONT_DATA_2 = array(
+		'name'       => 'Arial',
+		'slug'       => 'arial',
+		'fontFamily' => 'Arial',
+	);
+
+	function test_get_data() {
+		$font = new WP_Font_Family( self::FONT_DATA_1 );
+		$this->assertEquals( 'piazzolla', $font->get_data()['slug'] );
+		$this->assertEquals( 1, count( $font->get_data()['fontFace'] ) );
+	}
+
+	function test_get_data_as_json() {
+		$font = new WP_Font_Family( self::FONT_DATA_1 );
+		$this->assertEquals( wp_json_encode( self::FONT_DATA_1 ), $font->get_data_as_json() );
+	}
+
+	function test_has_font_faces() {
```

Comment:
Let's use a data provider here to ensure both scenarios are tested even if the first fails.

```php
/**
 * @dataProvider data_has_font_faces
 *
 * @param array $font_family An array of font family data.
 * @param bool  $expected    Whether the font family has font faces.
 */
public function test_has_font_faces( $font_family, $expected ) {
	$this->assertSame( $expected, $font_family->has_font_faces() );
}

/**
 * Data provider.
 *
 * @return array[]
 */
public function data_has_font_faces() {
	return array(
		'a font family with font faces'    => array(
			'font_family' => self::FONT_DATA_1,
			'expected'    => true,
		),
		'a font family without font faces' => array(
			'font_family' => self::FONT_DATA_2,
			'expected'    => false,
		),
	);
}

Reply:
Thanks for the example! Done